lib/ringhom.gi

```diff
@@ -557,18 +559,17 @@ function( R, I )
     R2:=Range(hom);
     I:=Subring(R2,List(GeneratorsOfRing(I),x->Image(hom,x)));
     nat:=NaturalHomomorphismByIdeal(R2,I);
-    return RingHomomorphismByImages(R,Range(nat),GeneratorsOfRing(R),
-             List(GeneratorsOfRing(R),x->Image(nat,Image(hom,i))));
+    return RingHomomorphismByImages(R,Range(nat),Rgens,
```

Comment:
Perhaps if this `NaturalHomomorphismByIdeal` notices that `IsRingWithOne(R)` is true, it could use this to make the quotient a ring-with-one, too, and set its one element?